bin/release.mjs

```diff
@@ -0,0 +1,259 @@
+          },
+          {
+            title: `Minor: v${nextMinor}`,
+            value: nextMinor,
+          },
+          {
+            title: `Major: v${nextMajor}`,
+            value: nextMajor,
+          },
+          {
+            title: "Custom",
+            value: null,
+          },
+        ],
+      },
+      { onCancel },
+    ));
+  }
+  if (!nextVersion) {
+    ({ nextVersion } = await prompts(
+      {
+        type: "text",
+        name: "nextVersion",
+        message: isValidSemver ? "Custom Version" : "Version",
+        validate: (value) => {
+          if (!value.trim()) return "Version is required";
+
+          return true;
+        },
+      },
+      { onCancel },
+    ));
+  }
+
+  nextVersion = nextVersion.trim();
+  const isNewValidSemver = Boolean(semver.valid(nextVersion));
+  if (isValidSemver) {
+    // If the existing version is not semver, don't bother with semver checks
+    if (isNewValidSemver) {
+      if (semver.lte(nextVersion, version)) {
+        await confirmOrExit(`Version ${nextVersion} is not greater than ${version}. Continue?`);
+      }
+      const cleaned = semver.clean(nextVersion);
+      if (cleaned !== nextVersion) {
+        let { clean } = await prompts({
+          type: "confirm",
+          name: "clean",
+          message: `Convert ${nextVersion} to cleaned version ${cleaned}?`,
+          initial: true,
+        });
+        if (clean) nextVersion = cleaned;
+      }
+    } else {
+      await confirmOrExit(`Version ${nextVersion} is not a valid semver. Continue?`, true);
+    }
+  }
+
+  // Update manifest.json and package.json
+  manifest.version = nextVersion;
+  packageJson.version = nextVersion;
+
+  // Write manifest.json and package.json (indent with 2 spaces and keep trailing newline)
+  writeFileSync(manifestPath, `${JSON.stringify(manifest, null, 2)}\n`);
+  writeFileSync(packagePath, `${JSON.stringify(packageJson, null, 2)}\n`);
+
+  // Stage changes
+  runCommand("git add manifest.json package.json");
+
+  // Commit changes
+  const { message } = await prompts(
+    {
+      type: "text",
+      name: "message",
+      message: "Commit message",
+      initial: `Release v${nextVersion}`,
+      validate: (value) => {
+        if (!value.trim()) return "Commit message is required";
+
+        return true;
+      },
+    },
+    { onCancel },
+  );
+
+  // Pick tag name
+  const existingTags = runCommand("git tag --list").split("\n").filter(Boolean);
+
+  const { tagName } = await prompts(
+    {
+      type: "text",
+      name: "tagName",
+      message: "Tag name",
+      initial: `v${nextVersion}`,
+      validate: (value) => {
+        if (!value.trim()) return "Tag name is required";
+
+        if (existingTags.includes(value)) return `Tag ${value} already exists`;
+
+        return true;
+      },
+    },
+    { onCancel },
+  );
+
+  // Commit changes
+  runCommand(`git commit -m "${message}"`);
+
+  // Tag commit
+  runCommand(`git tag ${tagName}`);
```

Comment:
Allow signing the tag. I know this can be set as a global git config, but I'd prefer to have it be passable as well. The flag is `-s`.